src/backend/services/request_validators.py

```diff
@@ -3,11 +3,7 @@
 from fastapi import HTTPException, Request
 
 import backend.crud.user as user_crud
-from backend.config.deployments import (
-    AVAILABLE_MODEL_DEPLOYMENTS,
-    find_config_by_deployment_id,
-    find_config_by_deployment_name,
-)
+from backend.config.deployments import AVAILABLE_MODEL_DEPLOYMENTS
```

Comment:
AVAILABLE_MODEL_DEPLOYMENTS is a list now. Could you please change this check in line 168
and deployment_name not in AVAILABLE_MODEL_DEPLOYMENTS.keys()

Reply:
I was able to substantially simplify the deployment header validation function by relying on the new deployments service